pkg/controllers/et/verticadbref_reconciler_test.go

```diff
@@ -0,0 +1,156 @@
+
+		Expect(k8sClient.Create(ctx, et)).Should(Succeed())
+		defer func() { Expect(k8sClient.Delete(ctx, et)).Should(Succeed()) }()
+
+		Expect(etRec.Reconcile(ctx, ctrl.Request{NamespacedName: et.ExtractNamespacedName()})).Should(Equal(ctrl.Result{}))
+
+		nm := et.ExtractNamespacedName()
+		etrigger := getEventTriggerStatus(ctx, nm)
+		Expect(etrigger.Status.References).ShouldNot(BeNil())
+		Expect(etrigger.Status.References[0].JobName).Should(BeEmpty())
+		Expect(etrigger.Status.References[0].JobNamespace).Should(BeEmpty())
+	})
+
+	It("should succeed with no-op when reference status job exists already exists", func() {
+		vdb := vapi.MakeVDB()
+		test.CreateVDB(ctx, k8sClient, vdb)
+		defer test.DeleteVDB(ctx, k8sClient, vdb)
+
+		et := vapi.MakeET()
+		status := vapi.ETRefObjectStatus{
+			Namespace: et.Spec.References[0].Object.Namespace,
+			Name:      et.Spec.References[0].Object.Name,
+			Kind:      et.Spec.References[0].Object.Kind,
+			JobName:   "test",
+		}
+
+		Expect(k8sClient.Create(ctx, et)).Should(Succeed())
+		defer func() { Expect(k8sClient.Delete(ctx, et)).Should(Succeed()) }()
+		Expect(etstatus.Apply(ctx, k8sClient, et, &status)).Should(Succeed())
+		Expect(etRec.Reconcile(ctx, ctrl.Request{NamespacedName: et.ExtractNamespacedName()})).Should(Equal(ctrl.Result{}))
+
+		nm := et.ExtractNamespacedName()
+		etrigger := getEventTriggerStatus(ctx, nm)
+		Expect(etrigger.Status.References).ShouldNot(BeNil())
+	})
+
+	It("should succeed with no-op when creating the job", func() {
+		vdb := vapi.MakeVDB()
+		test.CreateVDB(ctx, k8sClient, vdb)
+		defer test.DeleteVDB(ctx, k8sClient, vdb)
+
+		cond := []vapi.VerticaDBCondition{
+			{Type: vapi.AutoRestartVertica, Status: corev1.ConditionTrue},
+			{Type: vapi.DBInitialized, Status: corev1.ConditionTrue},
+		}
+		Expect(setVerticaStatus(ctx, k8sClient, vdb, cond)).Should(Succeed())
+
+		et := vapi.MakeET()
+		nm := et.ExtractNamespacedName()
+		Expect(k8sClient.Create(ctx, et)).Should(Succeed())
+		defer func() { Expect(k8sClient.Delete(ctx, et)).Should(Succeed()) }()
+		Expect(etRec.Reconcile(ctx, ctrl.Request{NamespacedName: nm})).Should(Equal(ctrl.Result{}))
+
+		job := makeJob(et)
+		etrigger := getEventTriggerStatus(ctx, nm)
+		Expect(etrigger.Status.References).ShouldNot(BeNil())
+		Expect(etrigger.Status.References[0].JobName).Should(Equal(et.Spec.Template.Metadata.Name))
+		Expect(etrigger.Status.References[0].JobNamespace).Should(Equal(et.Namespace))
+		defer func() { Expect(k8sClient.Delete(ctx, job)).Should(Succeed()) }()
+	})
+})
+
+func getEventTriggerStatus(ctx context.Context, nm types.NamespacedName) vapi.EventTrigger {
+	etrigger := vapi.EventTrigger{}
+	etStatus := k8sClient.Get(ctx, nm, &etrigger)
+	Expect(etStatus).Should(Succeed())
+
+	return etrigger
+}
+
+func setVerticaStatus(ctx context.Context, clnt client.Client, vdb *vapi.VerticaDB, conditions []vapi.VerticaDBCondition) error {
+	return vdbstatus.Update(ctx, clnt, vdb, func(vdb *vapi.VerticaDB) error {
```

Comment:
can you use `vdbstatus.UpdateCondition` for this? I know I said otherwise when you asked. But now I get the full context. The test that calls into here fails for me, and I think using `UpdateCondition` will resolve that.